src/Reflow/Parser.hs

```diff
@@ -31,7 +31,10 @@ header = do
     return $ Header $ n <> v
 
 headerName :: Parser Text
-headerName = pack <$> (many1 $ noneOf ":\n")
+headerName = do
+    n <- manyTill (noneOf "\n") (lookAhead $ string ":")
```

Comment:
I thought header names had to be in a limited character range. If so, this could be something like:

```hs
fmap pack $ (<>)
    <$> many1 $ oneOf $ '-':['0'..'Z']
    <*> string ":"
```

Do you actually want the trailing `":"` in the header name? If not, this could shorten to:

```hs
many1 (oneOf $ '-':['0'..'Z']) <* string ":"
```